src/Sulu/Bundle/MediaBundle/Controller/MediaController.php

```diff
@@ -101,85 +103,102 @@ function ($id) use ($locale, $mediaManager) {
-            }
+        $list = new ListRepresentation(
+            $listResponse,
+            self::$entityKey,
+            'cget_media',
+            $request->query->all(),
+            $listBuilder->getCurrentPage(),
+            $listBuilder->getLimit(),
+            $listBuilder->count()
+        );
+
+        return $this->handleView($this->view($list, 200));
+    }
 
-            $mediaManager = $this->getMediaManager();
+    /**
+     * Returns a list-builder for media list.
+     *
+     * @param Request $request
+     * @param FieldDescriptorInterface[] $fieldDescriptors
+     *
+     * @return DoctrineListBuilder
+     */
+    private function getListBuilder(Request $request, array $fieldDescriptors)
+    {
+        $restHelper = $this->get('sulu_core.doctrine_rest_helper');
+        $factory = $this->get('sulu_core.doctrine_list_builder_factory');
+        $listBuilder = $factory->create(self::$entityName);
+        $restHelper->initializeListBuilder($listBuilder, $fieldDescriptors);
+
+        // default sort by created
+        if (!$request->get('sortBy')) {
+            $listBuilder->sort($fieldDescriptors['created'], 'desc');
+        }
 
-            if ($ids !== null) {
-                $ids = explode(',', $ids);
-                $media = $mediaManager->getByIds($ids, $this->getLocale($request));
-            } else {
-                $media = $mediaManager->get(
-                    $this->getLocale($request),
-                    [
-                        'collection' => $collectionId,
-                        'types' => $types,
-                        'search' => str_replace('*', '%', $search),
-                        'orderBy' => $orderBy,
-                        'orderSort' => $orderSort,
-                        'systemCollections' => $this->getSecurityChecker()->hasPermission(
-                            'sulu.media.system_collections',
-                            PermissionTypes::VIEW
-                        ),
-                    ],
-                    $limit,
-                    $offset
+        $collectionId = $request->get('collection');
+        if ($collectionId) {
+            $collection = $this->getCollectionManager()->getById($collectionId, $this->getLocale($request));
+            if ($collection->getType()->getKey() === SystemCollectionManagerInterface::COLLECTION_TYPE) {
+                $this->getSecurityChecker()->checkPermission(
+                    'sulu.media.system_collections',
+                    PermissionTypes::VIEW
                 );
             }
+            $listBuilder->addSelectField($fieldDescriptors['collection']);
+            $listBuilder->where($fieldDescriptors['collection'], $collectionId);
+        }
 
-            $all = $mediaManager->getCount();
+        // If no limit is set in request and limit is set by ids
+        $requestLimit = $request->get('limit');
+        $ids = array_filter(explode(',', $request->get('ids')));
+        $numberIds = count($ids);
```

Comment:
Would name that `idsCount` or something like that, number sounds like they are casted into another type 😕